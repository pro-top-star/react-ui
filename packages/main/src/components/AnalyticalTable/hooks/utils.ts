export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
