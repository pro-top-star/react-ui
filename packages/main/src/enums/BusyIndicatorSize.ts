export enum BusyIndicatorSize {
  Small = 'Small',
  Medium = 'Medium',
  Large = 'Large'
}
