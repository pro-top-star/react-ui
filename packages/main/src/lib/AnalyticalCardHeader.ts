import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
