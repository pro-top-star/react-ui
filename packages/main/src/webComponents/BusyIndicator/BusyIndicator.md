[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/BusyIndicator)

```jsx
import { BusyIndicator } from '@ui5/webcomponents-react/lib/BusyIndicator';
```
