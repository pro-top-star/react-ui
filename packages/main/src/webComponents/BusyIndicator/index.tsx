import { BusyIndicatorSize } from '@ui5/webcomponents-react/lib/BusyIndicatorSize';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5BusyIndicator from '@ui5/webcomponents/dist/BusyIndicator';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface BusyIndicatorPropTypes extends WithWebComponentPropTypes {
  size?: BusyIndicatorSize; // @generated
  active?: boolean; // @generated
}

/**
 * <code>import { BusyIndicator } from '@ui5/webcomponents-react/lib/BusyIndicator';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/BusyIndicator" target="_blank">UI5 Web Components Playground</a>
 */
const BusyIndicator: FC<BusyIndicatorPropTypes> = withWebComponent<BusyIndicatorPropTypes>(UI5BusyIndicator);

BusyIndicator.displayName = 'BusyIndicator';

BusyIndicator.defaultProps = {
  size: BusyIndicatorSize.Large // @generated
};

export { BusyIndicator };
