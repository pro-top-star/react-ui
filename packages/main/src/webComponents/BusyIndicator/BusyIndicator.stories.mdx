import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { BusyIndicator } from '@ui5/webcomponents-react/lib/BusyIndicator';
import { BusyIndicatorSize } from '@ui5/webcomponents-react/lib/BusyIndicatorSize';
import { Select } from '@ui5/webcomponents-react/lib/Select';
import { Text } from '@ui5/webcomponents-react/lib/Text';
import { CSSProperties } from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';

<Meta
  title="UI5 Web Components / BusyIndicator"
  component={BusyIndicator}
  argTypes={{
    ...createSelectArgTypes({ size: BusyIndicatorSize }),
    children: { control: { disable: true } },
    slot: { control: { disable: true } },
    ref: { control: { disable: true } },
    style: {
      type: CSSProperties,
      description:
        'Element style which will be appended to the most outer element of a component. Use this prop carefully, some css properties might break the component.'
    },
    className: {
      type: 'string',
      description:
        'CSS Class Name which will be appended to the most outer element of a component. Use this prop carefully, overwriting CSS rules might break the component.'
    },
    tooltip: { type: 'string', description: 'A tooltip which will be shown on hover' }
  }}
  args={{
    size: BusyIndicatorSize.Medium,
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    active: true,
    ref: null
  }}
/>

<DocsHeader />

<Canvas>
  <Story name="Default">
    {(args) => {
      return <BusyIndicator {...args} />;
    }}
  </Story>
</Canvas>

<ArgsTable story="." />
<div style={{ fontFamily: 'var(--sapFontFamily)', fontSize: 'var(--sapFontSize)', color: 'var(--sapTextColor)' }}>
  <h3>Usage</h3>
  For the <code>BusyIndicator</code> you can define the size of the indicator, as well as whether it is shown or hidden.
  In order to hide it, use the html attribute <code>hidden</code> or <code>display: none;</code> <br />
  <br />
  In order to show busy state for an HTML element, simply nest the HTML element in a <code>BusyIndicator</code> instance.
  <br />
  <b>Note:</b> Since <code>BusyIndicator</code> has <code>display: inline-block;</code> by default and no width of its own,
  whenever you need to wrap a block-level element, you should set <code>display: block</code> to the busy indicator as well.
</div>

<br />
<br />

# Stories

<br />

## BusyIndicator with children

<Canvas>
  <Story name="With children" args={{ active: true }}>
    {(args) => {
      return (
        <BusyIndicator {...args}>
          <Select />
        </BusyIndicator>
      );
    }}
  </Story>
</Canvas>

<Canvas>
  <BusyIndicator active>
    <Text>
      Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et
      dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
      kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
      sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam
      voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata
      sanctus est Lorem ipsum dolor sit amet.
    </Text>
  </BusyIndicator>
</Canvas>
