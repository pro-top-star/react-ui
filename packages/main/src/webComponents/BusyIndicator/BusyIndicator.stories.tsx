import { boolean, select, text } from '@storybook/addon-knobs';
import { BusyIndicator } from '@ui5/webcomponents-react/lib/BusyIndicator';
import { BusyIndicatorSize } from '@ui5/webcomponents-react/lib/BusyIndicatorSize';
import React from 'react';

export default {
  title: 'UI5 Web Components / BusyIndicator',
  component: BusyIndicator
};

export const generatedDefaultStory = () => (
  <BusyIndicator
    active={boolean('active', true)}
    size={select('size', BusyIndicatorSize, BusyIndicatorSize.Medium)}
    text={text('text', 'Loading...')}
  >
    Some Content
  </BusyIndicator>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
