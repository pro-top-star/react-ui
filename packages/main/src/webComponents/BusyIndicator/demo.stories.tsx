import { boolean, select } from '@storybook/addon-knobs';
import { BusyIndicator } from '@ui5/webcomponents-react/lib/BusyIndicator';
import { BusyIndicatorSize } from '@ui5/webcomponents-react/lib/BusyIndicatorSize';
import React from 'react';

export default {
  title: 'UI5 Web Components / BusyIndicator',
  component: BusyIndicator
};

export const defaultStory = () => (
  <BusyIndicator size={select('size', BusyIndicatorSize, BusyIndicatorSize.Large)} active={boolean('active', true)} />
);

defaultStory.story = {
  name: 'Default story'
};
