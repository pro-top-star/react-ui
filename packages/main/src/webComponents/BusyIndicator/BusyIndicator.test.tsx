import { mountThemedComponent } from '@shared/tests/utils';
import { BusyIndicator } from '@ui5/webcomponents-react/lib/BusyIndicator';
import { BusyIndicatorSize } from '@ui5/webcomponents-react/lib/BusyIndicatorSize';
import React from 'react';

describe('BusyIndicator', () => {
  test('Basic Test', () => {
    const wrapper = mountThemedComponent(<BusyIndicator active />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('Inactive', () => {
    const wrapper = mountThemedComponent(<BusyIndicator />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('Size Medium', () => {
    const wrapper = mountThemedComponent(<BusyIndicator active size={BusyIndicatorSize.Medium} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('Size Small', () => {
    const wrapper = mountThemedComponent(<BusyIndicator active size={BusyIndicatorSize.Small} />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
