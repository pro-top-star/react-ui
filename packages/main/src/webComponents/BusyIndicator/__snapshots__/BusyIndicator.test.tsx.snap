// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusyIndicator Basic Test 1`] = `
<ui5-busyindicator
  active="true"
  size="Large"
/>
`;

exports[`BusyIndicator Inactive 1`] = `
<ui5-busyindicator
  size="Large"
/>
`;

exports[`BusyIndicator Size Medium 1`] = `
<ui5-busyindicator
  active="true"
  size="Medium"
/>
`;

exports[`BusyIndicator Size Small 1`] = `
<ui5-busyindicator
  active="true"
  size="Small"
/>
`;
